backend/infrahub/core/relationship/model.py

```diff
@@ -289,14 +289,21 @@ def get_peer_schema(self, db: InfrahubDatabase) -> MainSchemaTypes:
     def compare_properties_with_data(self, data: RelationshipPeerData) -> list[str]:
         different_properties = []
 
-        for prop_name, prop in data.properties.items():
-            if hasattr(self, "_flag_properties") and prop_name in self._flag_properties:
-                if prop.value != getattr(self, prop_name):
-                    different_properties.append(prop_name)
-
-            elif hasattr(self, "_node_properties") and prop_name in self._node_properties:
-                if prop.value != getattr(self, f"{prop_name}_id"):
-                    different_properties.append(prop_name)
```

Comment:
the old logic did not handle the case where a new property was in `self._node_properties` or `self._flag_properties`, but not in `data`